spark_ada/src/mlkem.adb

```diff
@@ -1964,35 +2056,20 @@ is
    end MLKEM_KeyGen;
 
 
-   function EK_Is_Valid_For_Encaps (EK : in MLKEM_Encapsulation_Key)
-     return Boolean
-   is
-      Key_To_Check : constant Poly_Zq_Vector_Bytes := EK (0 .. 384 * K - 1); --  calls _memcpy()
-      Decoded      : NTT_Poly_Zq_Vector;
-      Reencoded    : Poly_Zq_Vector_Bytes;
-   begin
-      --  FIPS 203 6.2 line 980 - 997
-      --    1. Check on the length of EK is a static type-check in SPARK, so
-      --       nothing to do here.
-      --    2. Modulus check - check that Decode/Encode is idempotent:
-      Decoded := ByteDecode12 (Key_To_Check);
-      Reencoded := ByteEncode12 (Decoded);
-      return Byte_Seq_Equal (Key_To_Check, Reencoded);
-   end EK_Is_Valid_For_Encaps;
-
-
+   -- This is also ML-KEM.Encaps_internal from FIPS 203 Algorithm 17
    procedure MLKEM_Encaps (EK       : in     MLKEM_Encapsulation_Key;
                            Random_M : in     Bytes_32;
-                           SS       :    out Bytes_32;
+                           Key      :    out Bytes_32;
```

Comment:
funny, I see you prefer `Key` to `SS`

Reply:
Again, in the spec they use "K" (Algorithm 20), which is overloaded, so I went for "Key"